modules/tensor_mechanics/src/materials/ComputeIsotropicElasticityTensor.C

```diff
@@ -66,7 +67,12 @@ ComputeIsotropicElasticityTensorTempl<is_ad>::ComputeIsotropicElasticityTensorTe
 
   if (_youngs_modulus_set && _youngs_modulus <= 0.0)
     mooseError("Youngs modulus must be positive in material '" + name() + "'.");
+}
 
+template <bool is_ad>
+void
+ComputeIsotropicElasticityTensorTempl<is_ad>::residualSetup()
+{
```

Comment:
If we use a function pointer, wouldn't it recompute every time it is called in computeQpProperties?

Reply:
I am thinking the function pointer would be called in the residualSetup. There would be two modes of operation, the current mode and the stochastic mode. 

In the current mode a function pointer (`_do_this_in_residual`) is initialized to an empty function. In stochastic mode, the pointer is initialized to a function that actually does the calculation as you are doing here. But, this is only needed if the speed hit is significant.

Obviously, in the current mode Cijkl just gets computed in the constructor by some function call.

Reply:
You only need to get this fancy if computeResidual is called a bunch, you can just do a boolean switch here as well.

When I started thinking about this I was thinking this was happening in computeQp... calls

Reply:
Ah, I see. How do we know if we are in stochastic mode? Is there some trigger in fe_problem? I was thinking of doing the boolean switch, but it might get complicated based on the shear number of parameters.  I'm probably worrying about this too much since this function is pretty trivial computationally.

Reply:
I think you are correct, this probably isn't that important with respect to time. If is it would probably just be a flag in the input parameters. Otherwise, we could look up if the parameter is being controlled, but that would need to happen on residualSetup and probably would take longer than doing the calc.